roles/commit_changes/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+
+- name: Assert required variables are defined
+  ansible.builtin.assert:
+    that:
+      - cc_directory is defined
+      - cc_directory | length > 0
+      - cc_message is defined
+      - cc_message | length > 0
+
+- name: Set git author name
+  ansible.builtin.command: git config user.name "{{ cc_author_name }}"
+  args:
+    chdir: "{{ cc_directory }}"
+  when:
+    - cc_author_name is defined
```

Comment:
If the variable is defined in the defaults file, I guess this check is not needed.